<!DOCTYPE html>
<html>
<head>
  <title>Reproducibility in life science</title>
  <meta charset="utf-8">
  <meta name="description" content="Reproducibility in life science">
  <meta name="author" content="Giusi Moffa">
  <meta name="generator" content="slidify" />
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta http-equiv="X-UA-Compatible" content="chrome=1">
  <link rel="stylesheet" href="libraries/frameworks/io2012/css/default.css" media="all" >
  <link rel="stylesheet" href="libraries/frameworks/io2012/css/phone.css" 
    media="only screen and (max-device-width: 480px)" >
  <link rel="stylesheet" href="libraries/frameworks/io2012/css/slidify.css" >
  <link rel="stylesheet" href="libraries/highlighters/highlight.js/css/solarized_light.css" />
  <base target="_blank"> <!-- This amazingness opens all links in a new tab. -->  <link rel=stylesheet href="./assets/css/ribbons.css"></link>

  
  <!-- Grab CDN jQuery, fall back to local if offline -->
  <script src="http://ajax.aspnetcdn.com/ajax/jQuery/jquery-1.7.min.js"></script>
  <script>window.jQuery || document.write('<script src="libraries/widgets/quiz/js/jquery.js"><\/script>')</script> 
  <script data-main="libraries/frameworks/io2012/js/slides" 
    src="libraries/frameworks/io2012/js/require-1.0.8.min.js">
  </script>
  
  

</head>
<body style="opacity: 0">
  <slides class="layout-widescreen">
    
    <!-- LOGO SLIDE -->
        <slide class="title-slide segue nobackground">
  <hgroup class="auto-fadein">
    <h1>Reproducibility in life science</h1>
    <h2>Reproducible reports with markdown, knitr, slidify</h2>
    <p>Giusi Moffa<br/>Practical Bioinformatics, 27 May 2014</p>
  </hgroup>
  <article></article>  
</slide>
    

    <!-- SLIDES -->
    <slide class="" id="slide-1" style="background:;">
  <article data-timings="">
    <!-- library(slidify) 
setwd("~/juicy/BioStatWork2012/mytex/journalClub/Reproducibility")
publish(user="annlia", repo="reproduce") ## for github
http://annlia.github.com/reproduce
publish(title = 'reproducibility', 'index.html', host = 'rpubs') ## for Rpubs
https://help.github.com/articles/generating-ssh-keys/
-->

<!-- pandoc index.md -t beamer -V theme:Warsaw -o bioInfo.pdf -->

<!-- setwd("~/juicy/BioStatWork2012/mytex/journalClub/Reproducibility/repBioInfo") -->

<!-- https://github.com/cboettig/knitcitations -->

<style>
em {
  font-style: italic
}
</style>

<style>
strong {
  font-weight: bold;
}
</style>

<!-- Why is reproducibility (and ideally replication) so important? -->

<h2>Casting doubt upon scientific findings</h2>

<ul>
<li><em>&quot;It ain&#39;t so much the things we don&#39;t know that get us into trouble. It&#39;s the things we know that just ain&#39;t so.&quot;</em> Uncertain source.</li>
</ul>

<p><a href="http://www.plosmedicine.org/article/info%3Adoi%2F10.1371%2Fjournal.pmed.0020124"><img src="figure/IoannidisPlosMedicine2005.jpg" width="600px" /></a><br>
Plos Medicine, 2005<br>
<a href="http://www.plosmedicine.org/article/related/info%3Adoi%2F10.1371%2Fjournal.pmed.0020124">Some links to its media coverage</a></p>

<ul>
<li><em>&quot;Basic research is like shooting an arrow in the air and , where it lands, painting a target.&quot;</em> Homer Adkins</li>
</ul>

<p><a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3596761/">This I believe in genetics: discovery can be a nuisance, replication is science, implementation matters</a><br>
John P. A. Ioannidis, 2013, frontiers in Genetics</p>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="class" id="id" style="background:;">
  <hgroup>
    <h2>Reproducibility in recent scientific publications</h2>
  </hgroup>
  <article data-timings="">
    <p><a href="http://www.sciencemag.org/content/334/6060/1225"><img src="figure/AgainAgainAgain.jpg" width="600px" /></a></p>

<p>Science, December 2011<br>
Barbara R. Jasny, Gilbert Chin, Lisa Chong, Sacha Vignieri  </p>

<p><a href="http://www.sciencemag.org/site/special/data-rep/"><img src="figure/scienceCover.gif" width="200px" /></a>
<img src="figure/Science2011Peng.jpg" width="500px" /></p>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-3" style="background:;">
  <hgroup>
    <h2>Science: again, and again, and again...</h2>
  </hgroup>
  <article data-timings="">
    <p>How do we promote the publication of replicable data?</p>

<h3>Scientific gold standard</h3>

<p><em>Replication:</em> the confirmation of results and conclusions from one study obtained independently in another.</p>

<h3>Difficulties</h3>

<ul>
<li>New tools and technologies</li>
<li>Massive amounts of data</li>
<li>Long-term studies</li>
<li>Interdisciplinary approaches</li>
<li>Complex questions</li>
</ul>

<h3>Attempts to replicate reveal scientific uncertainties</h3>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-4" style="background:;">
  <hgroup>
    <h2>Nature announces revised standards...</h2>
  </hgroup>
  <article data-timings="">
    <p><a href="http://www.nature.com/nature/focus/reproducibility/"><img src="figure/NatureIrreproducibility2013.png" width="600px" /></a> Special collection, April 2013</p>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-5" style="background:;">
  <hgroup>
    <h2>...and some picks from the Nature collection</h2>
  </hgroup>
  <article data-timings="">
    <ul>
<li><p><a href="http://www.nature.com/nature/journal/v487/n7408/full/487406a.html">Error prone</a><br>
Biologists must realize the pitfalls of work on massive amounts of data.<br>
Nature 487, 406 ( 26 July 2012 )</p></li>
<li><p><a href="http://www.nature.com/nature/journal/v483/n7391/full/483509a.html">Must try harder</a><br>
Too many sloppy mistakes are creeping into scientific papers. Lab heads must look more rigorously at the data — and at themselves.<br>
Nature 483, 509 ( 29 March 2012 )</p></li>
<li><p><a href="http://www.nature.com/news/if-a-job-is-worth-doing-it-is-worth-doing-twice-1.12727">If a job is worth doing, it is worth doing twice</a><br>
Researchers and funding agencies need to put a premium on ensuring that results are reproducible.<br>
Jonathan F. Russell<br>
Nature 496, 7 ( 04 April 2013 )</p></li>
<li><p><a href="http://www.nature.com/ng/journal/v41/n2/full/ng.295.html">Repeatability of published microarray gene expression analyses</a><br>
John P A Ioannidis et al,<br>
Nature Genetics 41, 2009</p></li>
</ul>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-6" style="background:;">
  <hgroup>
    <h2>...more picks from Nature</h2>
  </hgroup>
  <article data-timings="">
    <p><a href="http://www.nature.com/nature/journal/v497/n7450/full/497433a.html">Reproducibility: Six red flags for suspect work</a><br>
C. Glenn Begley explains how to recognize the preclinical papers in which the data won&#39;t stand up. Nature (23 May 2013)  </p>

<p><a href="http://www.nature.com/news/reproducibility-the-risks-of-the-replication-drive-1.14184">Reproducibility: The risks of the replication drive</a><br>
The push to replicate findings could shelve promising research and unfairly damage the reputations of careful, meticulous scientists, says Mina Bissell. Nature (20 November 2013)</p>

<p><img src="figure/SixRedFlags2013.jpg" height="300px" /> <img src="figure/RisksReplication2013.jpg" height="300px" /></p>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-7" style="background:;">
  <hgroup>
    <h2>The economist kicks in...</h2>
  </hgroup>
  <article data-timings="">
    <h3>Problems with scientific research</h3>

<p><a href="http://www.economist.com/news/leaders/21588069-scientific-research-has-changed-world-now-it-needs-change-itself-how-science-goes-wrong">How science goes wrong</a><br>
Scientific research has changed the world. Now it needs to change itself<br>
<a href="http://www.economist.com/news/briefing/21588057-scientists-think-science-self-correcting-alarming-degree-it-not-trouble">Trouble at the lab</a><br>
Scientists like to think of science as self-correcting. To an alarming degree, it is not<br>
Oct 19th 2013  </p>

<p><img src="figure/EconomistHowScienceGoesWrong.jpg" height="300px" /> <img src="figure/EconomistTroubleAtTheLab.jpg" height="300px" /></p>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-8" style="background:;">
  <hgroup>
    <h2>...and the New York Times</h2>
  </hgroup>
  <article data-timings="">
    <p><a href="http://www.nytimes.com/2014/01/21/science/new-truths-that-only-one-can-see.html">New Truths That Only One Can See.</a> George Johnson, JAN. 20, 2014<br>
<img src="figure/NYToneTruth.jpg" height="350px" /></p>

<p><a href="http://www.economist.com/node/21528593">The Duke saga:</a> <a href="http://projecteuclid.org/DPubS?service=UI&amp;version=1.0&amp;verb=Display&amp;handle=euclid.aoas/1267453942">Deriving chemosensitivity from cell lines: Forensic bioinformatics and reproducible research in high-throughput biology</a><br>
Ann. Appl. Stat., Volume 3, Number 4 (2009), 1309-1334.<br>
Keith A. Baggerly and Kevin R. Coombes. <a href="http://bioinformatics.mdanderson.org/Supplements/ReproRsch-All/Modified/">A starter set,</a> <a href="http://videolectures.net/cancerbioinformatics2010_baggerly_irrh/">Video lecture,</a> <a href="http://www.cbsnews.com/news/deception-at-duke-fraud-in-cancer-care/">60 Minutes</a></p>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-9" style="background:;">
  <hgroup>
    <h2>The statistics take...</h2>
  </hgroup>
  <article data-timings="">
    <p><a href="http://biostatistics.oxfordjournals.org/content/10/3/405.full">Reproducible research and Biostatistics</a><br>
Editorial 2009, Roger Peng</p>

<p><a href="http://biostatistics.oxfordjournals.org/content/15/1/1.full">An estimate of the science-wise false discovery rate and application to the top medical literature</a><br>
Leah R. Jager and Jeffrey T. Leek
Biostatistics (2014)<br>
with mixed discussions by top-class statisticians (including Yoav Benjamini, David R. Cox, Andrew Gelman, John P. A. Ioannidis)</p>

<ul>
<li><p><a href="http://andrewgelman.com/2014/02/20/differences-biology-statistics-explain-diverging-attitudes-regarding-criticism-replication-scientific-claims/">Do differences between biology and statistics explain some of our diverging attitudes regarding criticism and replication of scientific claims?</a></p></li>
<li><p><a href="http://simplystatistics.org/2013/12/16/a-summary-of-the-evidence-that-most-published-research-is-false/">A summary of the evidence that most published research is wrong.</a></p></li>
</ul>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-10" style="background:;">
  <hgroup>
    <h2>Some tools for reproducible research</h2>
  </hgroup>
  <article data-timings="">
    <ul>
<li>&quot;<em>Reproducibility is not an afterthought</em>&quot; <a href="http://biostatistics.oxfordjournals.org/content/11/3/385.full">(Donohue, 2010)</a>

<ul>
<li>plan it as soon you start a new project</li>
</ul></li>
<li>Golden rule: document everything! scripting your analysis, writing <em>&quot;everything&quot;</em> down (Roger Peng&#39;s analogy: scores of a symphony), 

<ul>
<li>in a text file (<em>&quot;future proof&quot;</em>)</li>
<li>a human readable format</li>
</ul></li>
<li>Literate programming (Donald Knuth), a stream of text + code chunks (e.g. <a href="http://yihui.name/knitr/"><code>knitr</code></a>), which can be <code>weaved</code> or <code>tangled</code>

<ul>
<li>documentation language (human readable)</li>
<li>programming language (machine readable)</li>
</ul></li>
</ul>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-11" style="background:;">
  <hgroup>
    <h2>Some tools for reproducible research</h2>
  </hgroup>
  <article data-timings="">
    <ul>
<li>Markup languages (\({\rm \LaTeX}\), Markdown, Html)</li>
<li>Integrated development environment <a href="https://www.rstudio.com/">(<code>RStudio</code>)</a></li>
<li>Cloud storage and version control, for storing and sharing</li>
<li>Unix-like shell programs</li>
<li>Save data in non-proprietary formats (again <em>&quot;future proof&quot;</em>)</li>
<li>The <code>analytic data</code> and the <code>source code</code> should be made available</li>
</ul>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-12" style="background:;">
  <hgroup>
    <h2>DOs and DON&#39;T (by Roger Peng)</h2>
  </hgroup>
  <article data-timings="">
    <ul>
<li>No! Graphical user interface (GUI) programs</li>
<li>No! Data cleaning by hands</li>
<li>No! Data download by clicking on links (you can teach the computer! <code>download.file</code> in <code>R</code>)</li>
<li>No! Point and click in the analysis</li>
<li>No! saving output, rather the steps of the analysis (e.g. preprocessing)</li>
</ul>

<p></br></p>

<ul>
<li>Yes! Teach a computer</li>
<li>Yes! Version control</li>
<li>Yes! Keep track of the software environment (Computer architecture, OS, softwares and packages, ...)</li>
<li>Yes! Set your seed (but with caution!)</li>
<li>Yes! Think of the entire pipeline (Raw data → processed data → analysis → report)</li>
</ul>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-13" style="background:;">
  <hgroup>
    <h2>Writing a reproducible report for getting in the news</h2>
  </hgroup>
  <article data-timings="">
    <p>So let&#39;s look at how we can get a dynamic and reproducible report with <code>knitr</code>, for getting in the news.</p>

<h3>The <code>knitr</code> process of an evolving document</h3>

<p>knittable document (Markup + code chunks)
  \(\longrightarrow\) Markup only 
    \(\longrightarrow\) Presentation</p>

<h3>Markdown example</h3>

<p><code>gettingTheNews.Rmd</code> 
  \(\longrightarrow\) <code>gettingTheNews.md</code> 
    \(\longrightarrow\) <code>gettingTheNews.html</code></p>

<h3>\({\rm \LaTeX}\) example</h3>

<p><code>gettingTheNews.Rnw</code> 
  \(\longrightarrow\) <code>gettingTheNews.tex</code> 
    \(\longrightarrow\) <code>gettingTheNews.pdf</code></p>

<p>Only ever edit the <code>.Rmd</code> or <code>.Rnw</code> files (intermediate changes will be lost when reprocessing them)</p>

<p><strong><code>!!!</code> The code needs to work for the document to be produced!</strong></p>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-14" style="background:;">
  <article data-timings="">
    <!-- ${\rm \LaTeX}$ does not work with pandoc -->

<h2>Markdown essentials</h2>

<h3>Headings</h3>

<pre><code>## I am a big header... 
### and I am a smaller one
</code></pre>

<h2>I am a big header...</h2>

<h3>and I am a smaller one</h3>

<h3>Italics and bold</h3>

<pre><code>*We are italic* and _We are italic_
**We are bold** and __We are bold__
</code></pre>

<p><em>We are italic</em> and <em>We are italic</em><br>
<strong>We are bold</strong> and <strong>We are bold</strong></p>

<p><code>Enter two spaces to go to a new line</code></p>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-15" style="background:;">
  <hgroup>
    <h2>Markdown essentials</h2>
  </hgroup>
  <article data-timings="">
    <h3>Unordered lists</h3>

<pre><code>Unordered List
* Things not to forget
* in no particular order
  * not me
  * and me
</code></pre>

<ul>
<li>Things not to forget</li>
<li>in no particular order

<ul>
<li>not me</li>
<li>and me</li>
</ul></li>
</ul>

<p></p>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-16" style="background:;">
  <hgroup>
    <h2>Markdown essentials</h2>
  </hgroup>
  <article data-timings="">
    <h3>Ordered List</h3>

<pre><code>1. I am the most important
2. Maybe I will be top soon
3. Running for the second
   * Won&#39;t be last
   * Anybody down there?
</code></pre>

<ol>
<li>I am the most important</li>
<li>Maybe I will be top soon</li>
<li>Running for the second

<ul>
<li>Won&#39;t be last</li>
<li>Anybody down there?</li>
</ul></li>
</ol>

<p></p>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-17" style="background:;">
  <hgroup>
    <h2>Markdown essentials</h2>
  </hgroup>
  <article data-timings="">
    <h3>Links</h3>

<pre><code>[The Markdown webpage](http://daringfireball.net/projects/markdown/)  
[The Github&#39;s Markdown guide](https://help.github.com/articles/github-flavored-markdown)
</code></pre>

<p><a href="http://daringfireball.net/projects/markdown/">The Markdown webpage</a><br>
<a href="https://help.github.com/articles/github-flavored-markdown">The Github&#39;s Markdown guide</a></p>

<h3>Code chunks</h3>

<pre><code>```{r mdLabel}
rnorm(10)
```
</code></pre>

<pre><code>##  [1] -0.3356  0.5986  0.3475 -0.8516  0.4719  1.2542 -0.5969  1.9572
##  [9] -0.9408 -1.2761
</code></pre>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-18" style="background:;">
  <hgroup>
    <h2>Markdown essentials</h2>
  </hgroup>
  <article data-timings="">
    <h3>\({\rm \LaTeX}\) equations and plots</h3>

<p><code>$\frac{1}{\sqrt{2\pi}\sigma}\exp(-\frac{(x-\mu)^2}{2\sigma^2})$</code>   \(\frac{1}{\sqrt{2\pi}\sigma}\exp(-\frac{(x-\mu)^2}{2\sigma^2})\)</p>

<pre><code class="rnorm">nS &lt;- 1000; y &lt;- data.frame(index = (1:nS), normS=rnorm(nS))
ggplot(y, aes(x=normS)) + geom_histogram(aes(y=..density..), binwidth=.2, 
  colour=&quot;darkblue&quot;, fill=&quot;white&quot;) + geom_density(alpha=.1, colour=joran, fill=&quot;#FFFF66&quot;,
    size=.8) + labs(y=&quot;&quot;, x=&quot;&quot;, title=&quot;Kernel density estimation&quot;) + 
      stat_function(fun = dnorm, colour = jpurp, size=.8)
</code></pre>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-19" style="background:;">
  <hgroup>
    <h2>Markdown essentials</h2>
  </hgroup>
  <article data-timings="">
    <h3>Syntax for including local images</h3>

<p><strong>Markdown:</strong> <code>![title](path/to/image)</code>, <strong>HTML:</strong> <code>&lt;img src=&quot;path/to/image&quot; /&gt;</code></p>

<p>Writing a <a href="http://www.theatlantic.com/entertainment/archive/2012/06/6-rules-for-a-great-story-inspired-by-snoopy/258413/">cool story</a></p>

<pre><code>&lt;img src=&quot;figure/snoobybanner.jpg&quot; height=&quot;250px&quot;&quot; /&gt;
</code></pre>

<p><img src="figure/snoobybanner.jpg" height="250px"" /></p>

<p><strong>Essential</strong>, but sufficient in most situations... and pretty amazing for drafting!</p>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-20" style="background:;">
  <hgroup>
    <h2>The Markdown news document</h2>
  </hgroup>
  <article data-timings="">
    <p>Make a new <code>.Rmd</code> file, here <code>gettingTheNews.Rmd</code>, the beginning of which looks like this:</p>

<pre><code>**Giusi Moffa**  
**Statistical bioinformatics, University of Regensburg**  
**Practical bioinformatics, 27 May 2014**  

p-values, one big topic for reproducible findings
========================================================

So let&#39;s look at how we can get a dynamic and reproducible 
report with `knitr`, for getting in the news.

</code></pre>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-21" style="background:;">
  <article data-timings="">
    <pre><code>
### Recording the $R$ session
It may be useful since different versions in general will not produce identical results.

    ```{r}
    # Print R session info
    sessionInfo()
    ```

It is a good habit also to set your working directory and check whether you are 
in the right place.

    ```{r}
    setwd(&quot;~/juicy/BioStatWork2012/mytex/journalClub/Reproducibility/repBioInfo&quot;)
    getwd() ## obtain the working directory
    ``` 

![Jelly Beans](figure/significant.png)    
</code></pre>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-22" style="background:;">
  <article data-timings="">
    <pre><code>## One old trick for getting in the news

Under the null-hypothesis p-values are expected to be uniformly distributed 
between 0 and 1. So if we have hundreds of crazy hypotheses we can try 
a fishing expedition for significance by testing all of them. 
On average 5% will come up as significant.

    ```{r, fig.height=3}
    set.seed(31)
    nColors &lt;- 200
    nObs &lt;- 100
    jellyBeans &lt;- matrix(rnorm(nObs*nColors), ncol=nColors)
    fishing4News &lt;- apply(jellyBeans, 2, function(x) t.test(x)$p.value)
    hist(fishing4News, col=4, freq=FALSE, main=&quot;Distribution of p-values&quot;, 
        xlab=&quot;p-values&quot;)
    lines(density(fishing4News), col=2, lwd=2)    
    ```
</code></pre>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-23" style="background:;">
  <hgroup>
    <h2>Create HTML and R files from Rmd</h2>
  </hgroup>
  <article data-timings="">
    <p>Compile a <code>html</code> file with <code>knit2html(&quot;gettingTheNews.Rmd&quot;)</code>  </p>

<p>Equivalent to</p>

<pre><code>knit(&quot;gettingTheNews.Rmd&quot;)
markdownToHTML(&quot;gettingTheNews.md&quot;)
</code></pre>

<p>The output will be <code>gettingTheNews.html</code></p>

<p>You can try to get a <code>pdf</code> from your markdown file with <a href="http://johnmacfarlane.net/pandoc/">pandoc</a></p>

<p>Extract <code>R</code> code as before with <code>purl</code></p>

<pre><code>purl(&quot;gettingTheNews.Rmd&quot;)
</code></pre>

<p>The output file will be <code>gettingTheNews.R</code></p>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-24" style="background:;">
  <hgroup>
    <h2>Back to the news via Markdown</h2>
  </hgroup>
  <article data-timings="">
    <p></br></p>

<h3>Recording the \(R\) session</h3>

<p>It may be useful since different versions in general will not produce identical results.</p>

<p></br></p>

<p>To be continued...</p>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-25" style="background:;">
  <article data-timings="">
    <pre><code class="r"># Print R session info
sessionInfo()
</code></pre>

<pre><code>## R version 3.3.0 (2016-05-03)
## Platform: x86_64-apple-darwin13.4.0 (64-bit)
## Running under: OS X 10.11.6 (El Capitan)
## 
## locale:
## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
## 
## attached base packages:
## [1] stats     graphics  grDevices utils     datasets  methods   base     
## 
## other attached packages:
## [1] knitr_1.14    ggplot2_2.1.0 slidify_0.5  
## 
## loaded via a namespace (and not attached):
##  [1] Rcpp_0.12.5      digest_0.6.10    mime_0.4         plyr_1.8.3      
##  [5] grid_3.3.0       gtable_0.2.0     formatR_1.4      magrittr_1.5    
##  [9] scales_0.4.0     evaluate_0.9     stringi_1.1.1    whisker_0.3-2   
## [13] rmarkdown_0.9.6  tools_3.3.0      stringr_1.1.0    munsell_0.4.3   
## [17] markdown_0.7.7   yaml_2.1.13      rsconnect_0.4.3  colorspace_1.2-6
## [21] htmltools_0.3.5
</code></pre>

<hr>

<h2>The news document</h2>

<p>It is a good habit also to set your working directory and check whether you are in the right place.</p>

<pre><code class="r">setwd(&quot;~/juicy/BioStatWork2012/mytex/journalClub/Reproducibility/repBioInfo&quot;)
getwd() ## obtain the working directory
</code></pre>

<pre><code>## [1] &quot;/Users/giusimoffa/juicy/BioStatWork2012/mytex/journalClub/Reproducibility/repBioInfo&quot;
</code></pre>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-26" style="background:;">
  <hgroup>
    <h2>p-values, one big topic for reproducible findings</h2>
  </hgroup>
  <article data-timings="">
    <p>Under the null-hypothesis p-values are expected to be uniformly distributed between 0 and 1. So if we have hundreds of crazy hypotheses we can try a fishing expedition for significance by testing all of them. On average 5% will come up as significant.</p>

<pre><code class="r">set.seed(31); nColors &lt;- 200; nObs &lt;- 100
jellyBeans &lt;- matrix(rnorm(nObs*nColors), ncol=nColors)
fishing4News &lt;- apply(jellyBeans, 2, function(x) t.test(x)$p.value)
hist(fishing4News, col=4, freq=FALSE, main=&quot;Distribution of p-values&quot;, xlab=&quot;p-values&quot;)
lines(density(fishing4News), col=2, lwd=2)
</code></pre>

<p><img src="assets/fig/unnamed-chunk-4-1.png" alt="plot of chunk unnamed-chunk-4"></p>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-27" style="background:;">
  <article data-timings="">
    <p>This way we get 7 significant results. This problem is well known in genomics and adressed by a plethora of methods for multiple correction. Transparency however is not always necessarily guaranteed, especially in in fields such as social science, and the issue is still a very much debated hot topic </p>

<ul>
<li><a href="http://www.stat.columbia.edu/%7Egelman/research/unpublished/p_hacking.pdf">The garden of forking paths: Why multiple comparisons can be a problem,
even when there is no <em>&quot;fishing expedition&quot;</em> or <em>&quot;\(p\)-hacking&quot;</em> and the research
hypothesis was posited ahead of time</a> (Gelman and Loken, 2013).</li>
<li><a href="http://www.stat.columbia.edu/%7Egelman/research/published/multiple2f.pdf">Why we (usually) don&#39;t need to worry about multiple comparisons</a> (Gelman et al, 2012).</li>
</ul>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-28" style="background:;">
  <hgroup>
    <h3>Correction for multiple testing</h3>
  </hgroup>
  <article data-timings="">
    <p>One way to account for multiple testing is by adopting the Benjamini-Hochberg correction. However if we are <em>&quot;lucky&quot;</em> we can still get in the news even if we correct for multiple testing.</p>

<pre><code class="r">set.seed(7)
nColors &lt;- 10000
nObs &lt;- 20
jellyBeans &lt;- matrix(rnorm(nObs*nColors), ncol=nColors)
fishing4News &lt;- apply(jellyBeans, 2, function(x) t.test(x)$p.value)
BHfishing &lt;- p.adjust(fishing4News, &quot;BH&quot;)
min(BHfishing)
</code></pre>

<pre><code>## [1] 0.01482826
</code></pre>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-29" style="background:;">
  <article data-timings="">
    <p>And if you really feel you need to, you can also save (and load) your entire workspace</p>

<pre><code>save.image(file=&quot;fishing.RData&quot;)
load(file=&quot;fishing.RData&quot;)
</code></pre>

<h3>Disclaimer</h3>

<p>This report is freely available for the benefit of <strong>science</strong>, so that our steps on the way to the news can be checked by anybody who wishes to do so.</p>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-30" style="background:;">
  <hgroup>
    <h2>Reproducing this presentation</h2>
  </hgroup>
  <article data-timings="">
    <h3>Installing and loading packages</h3>

<p>Install <code>knitr</code> (<code>markdown</code> should be installed by default) as usual, and <code>slidify</code> from <code>github</code> using the <code>devtools</code> package.</p>

<pre><code>install.packages(&quot;knitr&quot;)
</code></pre>

<pre><code>install.packages(&#39;devtools&#39;)
require(devtools)
pkgs &lt;- c(&quot;slidify&quot;, &quot;slidifyLibraries&quot;, &quot;rCharts&quot;)
install_github(pkgs, &#39;ramnathv&#39;, ref = &#39;dev&#39;)
</code></pre>

<p>Load packages and compile slides</p>

<pre><code>library(&#39;knitr&#39;); library(&#39;slidify&#39;)
setwd(&quot;~/path/to/repBioInfo&quot;) # edit ~/path/to/repBioInfo/index.Rmd
slidify(&quot;index.Rmd&quot;)
</code></pre>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-31" style="background:;">
  <hgroup>
    <h2>Making your own slidified presentation</h2>
  </hgroup>
  <article data-timings="">
    <h3>Knitting and slidifying</h3>

<ol>
<li>Write using R Markdown</li>
<li>Use an empty line followed by three dashes to separate slides!</li>
</ol>

<h3>Making your own deck</h3>

<pre><code>slidify(&quot;index.Rmd&quot;)
browseURL(&quot;index.html&quot;)
author(&quot;mydeck&quot;)
# edit ~/path/to/mydeck/index.Rmd
</code></pre>

<p>It can be easily shared with the <code>publish</code> command to <code>github</code>, <code>dropbox</code> or <code>Rpubs</code></p>

<pre><code>publish(&#39;myDeck&#39;, host = &quot;dropbox&quot;)
</code></pre>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-32" style="background:;">
  <hgroup>
    <h2>Making your own slidified presentation</h2>
  </hgroup>
  <article data-timings="">
    <p>The <code>author</code> command will produce a folder with the right scaffolding and a <code>index.Rmd</code> which you can edit as follows:</p>

<ol>
<li>Change the metadata in the YAML front matter to your taste.</li>
<li>Write your slides in R Markdown, separated by an empty line followed by three dashes.</li>
</ol>

<p>If <code>git</code> is installed it also performs necessary operation for initializing version control.</p>

<pre><code>title       : Reproducibility in life science
subtitle    : Reproducible reports with markdown, knitr, slidify
author      : Giusi Moffa
job         : Practical Bioinformatics, 27 May 2014
framework   : io2012       # {io2012, html5slides, shower, dzslides, ...}
highlighter : highlight.js  # {highlight.js, prettify, highlight}
hitheme     : solarized_light      # 
widgets     : [mathjax]            # {mathjax, quiz, bootstrap}
mode        : standalone # {selfcontained, standalone, draft}
</code></pre>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-33" style="background:;">
  <hgroup>
    <h2>More sophisticated documents</h2>
  </hgroup>
  <article data-timings="">
    <h3>\({\rm \LaTeX}\) + <code>knitr</code> \(\longrightarrow\) <code>Rnw</code> files, here <code>gettingTheNews.Rnw</code></h3>

<p>The equivalent of the markdown file will look as follows</p>

<pre><code>\documentclass{article}
\usepackage{hyperref}
\hypersetup{colorlinks,% 
                 urlcolor=blue}
\author{Giusi Moffa \\[2ex] Statistical bioinformatics, University of Regensburg\\[4ex]}
\title{A \texttt{knitr} + \LaTeX~ dynamic report.\\[4ex]}
\date{27 May 2014 \\[4ex] Practical bioinformatics}

\begin{document}
\maketitle
\section*{p-values, one big topic for reproducible findings}
So let&#39;s look at how we can get a dynamic and reproducible report with `knitr`, 
for getting in the news.
</code></pre>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-34" style="background:;">
  <article data-timings="">
    <pre><code>\subsubsection*{Recording the $R$ session.}
It may be useful since different versions in general will not produce identical results.

&lt;&lt;rInfo, results=&#39;asis&#39;&gt;&gt;=
# Print R session info
toLatex(sessionInfo())
@

It is a good habit also to set your working directory and check whether you are 
in the right place.

&lt;&lt;rWD&gt;&gt;=
setwd(&quot;~/juicy/BioStatWork2012/mytex/journalClub/Reproducibility/repBioInfo&quot;)
getwd() ## obtain the working directory
@

\subsubsection*{One old trick for getting in the news}
\includegraphics[width=.6\textwidth]{figure/significant.png}
%\clearpage

</code></pre>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-35" style="background:;">
  <article data-timings="">
    <pre><code>Under the null-hypothesis p-values are expected to be uniformly distributed 
between 0 and 1. So if we have hundreds of crazy hypotheses we can try a 
fishing expedition for significance by testing all of them. 
On average 5% will come up as significant.

&lt;&lt;rSimul, fig.height=3&gt;&gt;=
set.seed(31)
nColors &lt;- 200
nObs &lt;- 100
jellyBeans &lt;- matrix(rnorm(nObs*nColors), ncol=nColors)
fishing4News &lt;- apply(jellyBeans, 2, function(x) t.test(x)$p.value)
@
</code></pre>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-36" style="background:;">
  <hgroup>
    <h2>Create pdf and R files from Rnw</h2>
  </hgroup>
  <article data-timings="">
    <p>Compile a <code>pdf</code> file with <code>knit2pdf(&quot;gettingTheNews.Rnw&quot;)</code>  </p>

<p>Equivalent to</p>

<pre><code>knit(&quot;gettingTheNews.Rnw&quot;)
library(tools)
texi2pdf(&quot;gettingTheNews.tex&quot;)
</code></pre>

<p>The output will be <code>gettingTheNews.pdf</code></p>

<p>Extract <code>R</code> code with <code>purl</code></p>

<pre><code>purl(&quot;gettingTheNews.Rnw&quot;)
</code></pre>

<p>The output file will be <code>gettingTheNews.R</code></p>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-37" style="background:;">
  <hgroup>
    <h2>More knitr basics</h2>
  </hgroup>
  <article data-timings="">
    <h3>\({\rm \LaTeX}\) code chunks</h3>

<p>Started with &lt;&lt; &gt;&gt;= and ended with @</p>

<pre><code>&lt;&lt;rLabel, eval=FALSE&gt;&gt;=
  y &lt;- rnorm(1000)
  library(ggplot2)
  joran &lt;- rgb(.8,.4,0)
  jpurp &lt;- rgb(.5,0,1)
  hist(y, col=4)
@
</code></pre>

<ul>
<li>Inline code is inserted with <code>Sexpr{}</code></li>
</ul>

<p><a href="http://cran.r-project.org/web/packages/knitr/vignettes/knitr-refcard.pdf">knitr quick reference</a></p>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-38" style="background:;">
  <hgroup>
    <h2>A selection of chunk options</h2>
  </hgroup>
  <article data-timings="">
    <ul>
<li>eval - <code>FALSE</code> for no evaluation</li>
<li>echo - <code>FALSE</code> for not showing the code</li>
<li>results - <code>hide</code> for suppressing the output</li>
<li>include - <code>FALSE</code> for evaluating a chunk without including the output</li>
<li>tidy - <code>FALSE</code> for disabling formatting (useful if the code is not wrapping properly)</li>
<li>fig.height - to set the height of a figure</li>
<li>fig.width - to set the width of a figure</li>
<li>cache - <code>TRUE</code> for storing long calculations

<ul>
<li>any change, however small, will result in the chunk being re-evaluated</li>
<li>use <code>dependson</code> to set dependencies</li>
</ul></li>
<li><a href="http://yihui.name/knitr/options">more options</a><br></li>
<li><a href="http://yihui.name/knitr/hooks"><code>hooks</code></a> allow for more customization<br></li>
</ul>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-39" style="background:;">
  <hgroup>
    <h2>References and credits</h2>
  </hgroup>
  <article data-timings="">
    <p><a href="http://yihui.name/knitr/">knitr</a></p>

<p><a href="http://slidify.org/">slidify</a></p>

<p><a href="http://www.crcpress.com/product/isbn/9781482203530?source=igodigital"><img src="figure/DDRknitr.jpg" height="300px"" /></a> <a href="http://www.crcpress.com/product/isbn/9781466572843"><img src="figure/RRRRStudio.jpg" height="300px"" /></a> <a href="http://www.crcpress.com/product/isbn/9781466561595"><img src="figure/implementingRR.jpg" height="300px"" /></a></p>

<p><a href="http://biostats.bepress.com/bioconductor/paper2/">Statistical Analyses and Reproducible Research</a> (Gentleman and Lang, 2004)</p>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-40" style="background:;">
  <hgroup>
    <h2>Further (blog) readings</h2>
  </hgroup>
  <article data-timings="">
    <p><a href="http://magazine.amstat.org/blog/2011/01/01/scipolicyjan11/">Reproducible Research</a></p>

<p>Evidence-based Data Analysis: Treading a New Path for Reproducible Research:<br>
<a href="http://simplystatistics.org/2013/08/21/treading-a-new-path-for-reproducible-research-part-1/">Part 1,</a> <a href="http://simplystatistics.org/2013/08/28/evidence-based-data-analysis-treading-a-new-path-for-reproducible-research-part-2/">Part 2,</a> <a href="http://simplystatistics.org/2013/09/05/implementing-evidence-based-data-analysis-treading-a-new-path-for-reproducible-research-part-3/">Part 3</a> </p>

<p><a href="http://simplystatistics.org/2011/11/06/reproducible-research-notes-from-the-field/">Reproducible research: Notes from the field</a></p>

<p><a href="http://simplystatistics.org/2012/07/03/replication-and-validation-in-omics-studies-just-as/">Replication and validation in -omics studies - just as important as reproducibility</a></p>

<p><a href="http://kbroman.github.io/Tools4RR/">Tools for Reproducible Research, by Karl Broman</a></p>

<p><a href="http://kbroman.github.io/Tools4RR/pages/resources.html">Resources and further reading</a><br>
<a href="http://www.siam.org/news/news.php?id=2064">Top Ten Reasons To Not Share Your Code</a>, and why you should anyway. Randall J. LeVeque (Siam news, April 1, 2013)</p>

<p><a href="http://www.nature.com/news/2010/101013/full/467753a.html">Publish your computer code: it is good enough</a>. Nick Barnes (Nature, 13 October 2010)</p>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-41" style="background:;">
  <hgroup>
    <h2>From science to anecdotes... and viceversa</h2>
  </hgroup>
  <article data-timings="">
    <p><img src="figure/reproSpectrum.jpg" width="650px"" /> <img src="figure/Difference.png" height="550px"" /></p>

  </article>
  <!-- Presenter Notes -->
</slide>

    <slide class="backdrop"></slide>
  </slides>
  <div class="pagination pagination-small" id='io2012-ptoc' style="display:none;">
    <ul>
      <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=1 title='NA'>
         1
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=2 title='Reproducibility in recent scientific publications'>
         2
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=3 title='Science: again, and again, and again...'>
         3
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=4 title='Nature announces revised standards...'>
         4
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=5 title='...and some picks from the Nature collection'>
         5
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=6 title='...more picks from Nature'>
         6
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=7 title='The economist kicks in...'>
         7
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=8 title='...and the New York Times'>
         8
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=9 title='The statistics take...'>
         9
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=10 title='Some tools for reproducible research'>
         10
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=11 title='Some tools for reproducible research'>
         11
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=12 title='DOs and DON&#39;T (by Roger Peng)'>
         12
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=13 title='Writing a reproducible report for getting in the news'>
         13
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=14 title='NA'>
         14
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=15 title='Markdown essentials'>
         15
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=16 title='Markdown essentials'>
         16
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=17 title='Markdown essentials'>
         17
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=18 title='Markdown essentials'>
         18
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=19 title='Markdown essentials'>
         19
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=20 title='The Markdown news document'>
         20
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=21 title='NA'>
         21
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=22 title='NA'>
         22
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=23 title='Create HTML and R files from Rmd'>
         23
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=24 title='Back to the news via Markdown'>
         24
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=25 title='NA'>
         25
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=26 title='p-values, one big topic for reproducible findings'>
         26
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=27 title='NA'>
         27
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=28 title='Correction for multiple testing'>
         28
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=29 title='NA'>
         29
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=30 title='Reproducing this presentation'>
         30
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=31 title='Making your own slidified presentation'>
         31
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=32 title='Making your own slidified presentation'>
         32
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=33 title='More sophisticated documents'>
         33
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=34 title='NA'>
         34
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=35 title='NA'>
         35
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=36 title='Create pdf and R files from Rnw'>
         36
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=37 title='More knitr basics'>
         37
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=38 title='A selection of chunk options'>
         38
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=39 title='References and credits'>
         39
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=40 title='Further (blog) readings'>
         40
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=41 title='From science to anecdotes... and viceversa'>
         41
      </a>
    </li>
  </ul>
  </div>  <!--[if IE]>
    <script 
      src="http://ajax.googleapis.com/ajax/libs/chrome-frame/1/CFInstall.min.js">  
    </script>
    <script>CFInstall.check({mode: 'overlay'});</script>
  <![endif]-->
</body>
  <!-- Load Javascripts for Widgets -->
  
  <!-- MathJax: Fall back to local if CDN offline but local image fonts are not supported (saves >100MB) -->
  <script type="text/x-mathjax-config">
    MathJax.Hub.Config({
      tex2jax: {
        inlineMath: [['$','$'], ['\\(','\\)']],
        processEscapes: true
      }
    });
  </script>
  <script type="text/javascript" src="http://cdn.mathjax.org/mathjax/2.0-latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
  <!-- <script src="https://c328740.ssl.cf1.rackcdn.com/mathjax/2.0-latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML">
  </script> -->
  <script>window.MathJax || document.write('<script type="text/x-mathjax-config">MathJax.Hub.Config({"HTML-CSS":{imageFont:null}});<\/script><script src="libraries/widgets/mathjax/MathJax.js?config=TeX-AMS-MML_HTMLorMML"><\/script>')
</script>
<!-- LOAD HIGHLIGHTER JS FILES -->
  <script src="libraries/highlighters/highlight.js/highlight.pack.js"></script>
  <script>hljs.initHighlightingOnLoad();</script>
  <!-- DONE LOADING HIGHLIGHTER JS FILES -->
   
  </html>